store/sqlstore/post_store.go

```diff
@@ -484,9 +522,14 @@ type etagPosts struct {
 func (s *SqlPostStore) InvalidateLastPostTimeCache(channelId string) {
 }
 
-func (s *SqlPostStore) GetEtag(channelId string, allowFromCache bool) string {
+func (s *SqlPostStore) GetEtag(channelId string, allowFromCache, collapsedThreads bool) string {
+	q := s.getQueryBuilder().Select("Id", "UpdateAt").From("Posts").Where(sq.Eq{"ChannelId": channelId}).OrderBy("UpdateAt DESC").Limit(1)
+	if collapsedThreads {
```

Comment:
I don't fully understand why this is changing the Etag, this basically convert any  not-root-post in a not-etag-able data. The idea of the etag, is to not change if nothing has changed, and change if something has change. In this case, we need to find something that only changes is the expected result is changed. Maybe we can use something like the Max value between Post.UdpateAt and Thread.LastReplyAt, or something like that. Maybe that is not needed because it is supposed to not used the collapsed thread for not-root posts. I think this is a bet confusing, can work, but maybe we can add a comment with the intention.

Reply:
@jespino - My thinking was: ETag is used to cache results of a channel. We calculate etag based on UpdateAt. If we are in collapsed mode, the channel shows only root posts. Thus I added the query clause. If we keep taking UpdateAt from regular posts I think the caching will be less efficient. Thoughts?

Reply:
Yes, I think you are right, I think I miss understood the method itself, I though was about 1 post, and now I see is about the channel itself.